### PR TITLE
Use task_group::defer for final task

### PR DESCRIPTION
**Pull request by nsmith-**
Each Lane is given a task to run after it exhausts all events (in the recursive `doNextEvent` call). This task is registered in the group ahead of time to assure the group waits for it to be run (and hence, the Lane is done with all work)

### CODE REVIEW

File: threaded_io_test.cc [hunk 1]
```diff
@@ -10,11 +10,14 @@
 
 #include "CLI11.hpp"
 
+#define TBB_PREVIEW_TASK_GROUP_EXTENSIONS 1 // for task_group::defer
```
Comment (Dr15Jones) [inline threaded_io_test.cc:13]: Would be better to declare this in CMakeList file

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

File: Lane.cc [hunk 2]
```diff
@@ -61,17 +61,17 @@ void Lane::processEventAsync(tbb::task_group& group, TaskHolder iCallback, const
...
   presentEventIndex_ = index++;
   if(source_->mayBeAbleToGoToEvent(presentEventIndex_)) {
     if(verbose_) {
       std::cout <<"event "+std::to_string(presentEventIndex_)+"\n"<<std::flush;
     }
     
-    OptionalTaskHolder processEventTask(group, make_functor_task([this,&index, &group, &outputer, counter]() {
-          TaskHolder recursiveTask(group, make_functor_task([this, &index, &group, &outputer, counter]() {
-                doNextEvent(index, group, outputer, std::move(counter));
+    OptionalTaskHolder processEventTask(group, make_functor_task([this,&index, &group, &outputer, finalTask]() {
```
Comment (Dr15Jones) [inline Lane.cc:72]:
```suggestion
    OptionalTaskHolder processEventTask(group, make_functor_task([this,&index, &group, &outputer, finalTask=std::move(finalTask)]() {
```

File: Lane.cc [hunk 3]
```diff
@@ -61,17 +61,17 @@ void Lane::processEventAsync(tbb::task_group& group, TaskHolder iCallback, const
...
+          TaskHolder recursiveTask(group, make_functor_task([this, &index, &group, &outputer, finalTask]() {
```
Comment (Dr15Jones) [inline Lane.cc:73]:
```suggestion
          TaskHolder recursiveTask(group, make_functor_task([this, &index, &group, &outputer, finalTask=std::move(finalTask)]() {
```

File: Lane.cc (hunk 2 above)
Comment (nsmith-) [reply Lane.cc:72]: I assumed there was a reason the counter wasn't moved, but it seems ok

**Comment by Dr15Jones** [comment]
@nsmith- thanks for the change!

**Pull request merged**